ios/Classes/NativeTagTechnologyProxy.m

```diff
@@ -0,0 +1,40 @@
+/**
+ * Axway Titanium
+ * Copyright (c) 2009-present by Axway Appcelerator. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+
+#import "NativeTagTechnologyProxy.h"
+#import "TiBlob.h"
+#import "TiNFCTagProxy.h"
+#import "TiNfcUtilities.h"
+#import "TiUtils.h"
+
+@implementation NativeTagTechnology
+
+#pragma mark Public API's
+
+- (void)connect:(id)unused
+{
+  [session connectToTag:tag
+      completionHandler:^(NSError *_Nullable error) {
+        [self fireEvent:@"didConnectTag"
```

Comment:
I think before calling event, you should also check if event is set or not.
```
if (![self _hasListeners:@"didConnectTag"]) {
    return;
  }
```

Reply:
Updated accordingly.